subprojects/gradle/instrumentation-test-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/metadata/ModulePath.kt

```diff
@@ -0,0 +1,11 @@
+package com.avito.instrumentation.impact.metadata
+
+data class ModulePath(val path: String) {
+    init {
+        require(path.matches(validModulePath)) { "$path is not a valid gradle module path" }
+    }
+}
+
+fun ModulePath.toFileName() = path.drop(1).replace(":", "+")
+
+private val validModulePath = Regex("(:[a-zA-Z1-9\\-]+(_[a-zA-Z1-9\\-])?)*")
```

Comment:
```suggestion
private val validModulePath = Regex("(:[\w\d\\-]+(_[\w\d\\-])?)*")
```
It's good to use character classes for readability.

Reply:
Yet another great part to delegate to Gradle

Reply:
As far as I see, it's a custom format similar to the gradle path.

Reply:
used gradle one